test/infrastructure/docker/go.mod

```diff
@@ -9,18 +9,18 @@ require (
 	github.com/onsi/gomega v1.7.1
 	github.com/pkg/errors v0.8.1
 	k8s.io/api v0.0.0-20191121015604-11707872ac1c
-	k8s.io/apiextensions-apiserver v0.0.0-20190918201827-3de75813f604 // indirect
 	k8s.io/apimachinery v0.0.0-20191121015412-41065c7a8c2a
 	k8s.io/client-go v11.0.0+incompatible
 	k8s.io/klog v1.0.0
 	sigs.k8s.io/cluster-api v0.2.6-0.20191216153338-db9baf9ade0c
 	sigs.k8s.io/cluster-api/test/framework v0.0.0-20191216153338-db9baf9ade0c
 	sigs.k8s.io/controller-runtime v0.4.0
-	sigs.k8s.io/kind v0.5.1
+	sigs.k8s.io/kind v0.6.1
 )
 
 replace (
 	k8s.io/client-go => k8s.io/client-go v0.0.0-20190918160344-1fbdaa4c8d90
 	sigs.k8s.io/cluster-api => ../../..
 	sigs.k8s.io/cluster-api/test/framework => ../../framework
+	sigs.k8s.io/kind => sigs.k8s.io/kind v0.5.1
```

Comment:
Why not use v0.6.x?

Reply:
I'll take a look at this version dependency. When I initially started on this work I was running into an issue with v0.6.x. However, that might be different as I've been rebasing master onto this branch and it could've changed.

Reply:
So in `test/infrastructure/docker` when I remove the replace directive I get the following error when `make modules`. I can do the refactor to support the newer `kind` package as part of a separate PR.
```
$ make modules
go mod tidy
sigs.k8s.io/cluster-api/test/infrastructure/docker/docker imports
        sigs.k8s.io/kind/pkg/container/docker: module sigs.k8s.io/kind@latest found (v0.6.1), but does not contain package sigs.k8s.io/kind/pkg/container/docker
sigs.k8s.io/cluster-api/test/infrastructure/docker/e2e imports
        sigs.k8s.io/kind/pkg/container/cri: module sigs.k8s.io/kind@latest found (v0.6.1), but does not contain package sigs.k8s.io/kind/pkg/container/cri
make: *** [modules] Error 1
```

Reply:
the library completely changed in kind v0.6, there's an open issue and a bunch of work that needs to be done to use the new library. It is a large amount of work.